opal/corelib/array.rb

```diff
@@ -10,6 +10,16 @@ class Array < `Array`
   `Opal.defineProperty(self.$$prototype, '$$is_array', true)`
 
   %x{
+    function shiftNoArg(list) {
```

Comment:
Can you add a mention about the performance regression in a comment before the function definition?
I'm thinking about our future selves… and how we might want to remove the optimization if V8 goes back to optimizing the vanilla version.

Reply:
Sure, actually there are quite a few things to mention. I will also add a link to the original implementation (found in the Node.js codebase)

Reply:
@elia Done